app/Services/UpdateEmailService.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brewmap\Services;
+
+use Brewmap\Eloquent\User;
+use Brewmap\Exceptions\User\NewEmailChangingException;
+use Brewmap\Notifications\Mail\EmailChangeNotification;
+use Illuminate\Notifications\AnonymousNotifiable;
+
+class UpdateEmailService
+{
+    protected AnonymousNotifiable $anonymousNotifiable;
+
+    public function __construct(AnonymousNotifiable $anonymousNotifiable)
+    {
+        $this->anonymousNotifiable = $anonymousNotifiable;
+    }
+
+    public function notifyAboutNewEmail(string $email, string $id): void
```

Comment:
Why `notifyAboutNewEmail()` and `updateEmail()` methods are taking the same arguments but in different order? And why yo're expecting string id? Maybe `User` instance would be better?

Reply:
I changed that like this. I pass **User** to the service function and I am getting id form User instance. It looks beeter.
```
public function notifyAboutNewEmail(User $user, string $email): void
    {
        $this->anonymousNotifiable
            ->route("mail", $email)
            ->notify(new EmailChangeNotification($user->id));
    }
```